dll/win32/shell32/folders/CDesktopFolder.cpp

```diff
@@ -1069,6 +1069,13 @@ HRESULT WINAPI CDesktopFolderViewCB::MessageSFVCB(UINT uMsg, WPARAM wParam, LPAR
         case SFVM_VIEWRELEASE:
             m_pShellView = NULL;
             return S_OK;
+        case SFVM_GETCOMMANDDIR:
+        {
+            WCHAR buf[MAX_PATH];
+            if (SHGetSpecialFolderPathW(NULL, buf, CSIDL_DESKTOPDIRECTORY, TRUE))
```

Comment:
Isn't it possible to use the already cached `sPathTarget` ? (but that one belongs to the `CDesktopFolder` class.)

Reply:
In theory, yes but sPathTarget is:

- Not part of this class and holding a reference to CDesktopFolder could be problematic and more complicated. 
- sPathTarget as a concept is strange because the desktop has two filesystem folders. Take a look at `_GetSFFromPidl` for example, if you have a file with the same name in both desktop folders, you could only ever access the users file because it checks with GetFileAttributes. Windows of course does not work like this, it encodes which folder a pidl is for inside the pidl somehow. (I will file a bug for this).

Reply:
Yes, this merged view is what I'm talking about.

It also applies to the Start menu but there is no specific folder for that in shell32.